### DISCUSSION TITLE
Does MRTK works with the new Unity InputSystem package?

### DISCUSSION THREAD

**Discussion by binxie33** [microsoft/MixedRealityToolkit-Unity#10360] [Q&A] [answered]
I am trying to create a custom InputDevice that conforms to the Unity InputSystem package 

https://docs.unity3d.com/Packages/com.unity.inputsystem@1.0/manual/index.html 
https://docs.unity3d.com/Packages/com.unity.inputsystem@1.0/manual/Devices.html

Will it work with MRTK?

**Comment by david-c-kline** [comment, marked as answer]
@binxie33, 

In MRTK v2, we have no usage of the new Unity Input System package.

In the upcoming MRTK v3, it will be based on the new input system.

**Reply by keveleigh** [reply]
To follow-on, if you want to use the new input system in your own app, please just be sure to leave the active input system dropdown set to "both"
![image](https://user-images.githubusercontent.com/3580640/145267358-6322c193-ff1b-4b69-83bc-a1fad768e4e5.png)
.

As @davidkline-ms mentioned, MRTK 3 is almost entirely built on the new input system

**Reply by binxie33** [reply]
Thank you for the information. My project is a long-term project and will eventually use MRTK3, so knowing it will work with MRTK 3 is a great news already. Keep on the amazing work!!
